packages/tenant-process/src/services/tenantService.ts

```diff
@@ -469,6 +471,52 @@ export function tenantServiceBuilder(
       );
     },
 
+    async addCertifierId(
+      {
+        tenantId,
+        certifierId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        certifierId: string;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(`Adding certifierId to Tenant ${tenantId}`);
+
+      const tenant = await retrieveTenant(tenantId, readModelService);
+
+      if (
+        tenant.data.features.some(
+          (feature) => feature.certifierId === certifierId
```

Comment:
the error is triggered only if trying to assign the same certifierId. Is this the intended behaviour?
Or should it raise the error even if the certifierId is different? (for example if we are trying to change the id)

Reply:
Yes, this should be the behavior.
The check should be that if the certifierId is already present, you cannot add the certifierId because the tenant is already a certifier

Reply:
I may be wrong, but in Scala code this is not checked.
From my side it's ok to add it, but we should define its behaviour, among
- check if the same certifierId has already been assigned, as done in this PR. It blocks the creation of a duplicated event, but it does not prevent changing che certifierId (which would have impact on created certified attributes)
- check if the certifierId has already been assigned, with any value. This prevents the previous point, but it does not allow to change the value in case of mistake. Since the value is a manual input, I would probably avoid this solution
- a bit of both. Allow to set any value, but only if the tenant has not yet created a certified attribute. Slightly increase the complexity, and it may be impacted by eventual consistency, but more flexible

I would go for the third, but let's see if @beetlecrunch had other ideas

Reply:
We set this value and control it, but I like the third solution proposed by @galales.